src/main/scala/mesosphere/marathon/state/UpgradeStrategy.scala

```diff
@@ -23,4 +24,9 @@ object UpgradeStrategy {
     strategy.minimumHealthCapacity is between(0.0, 1.0)
     strategy.maximumOverCapacity is between(0.0, 1.0)
   }
+
+  lazy val validForResidentTasks: Validator[UpgradeStrategy] = validator[UpgradeStrategy] { strategy =>
+    strategy.minimumHealthCapacity should be <= 0.5
+    strategy.maximumOverCapacity should be <= 0.00001
```

Comment:
I learned to never use `==` on doubles. Scala implements equality on Double - I can use it.